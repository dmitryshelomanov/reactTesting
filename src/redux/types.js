export const COUNTER_PLUS = 'Counter/PLUS'
export const COUNTER_MINES = 'Counter/MINES'
