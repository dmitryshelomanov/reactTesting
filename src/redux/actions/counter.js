import { COUNTER_MINES, COUNTER_PLUS } from '../types'


export const plus = (num) => (dispatch) => {
  return dispatch({
    payload: num,
    type: COUNTER_PLUS,
  })
}

export const mines = (num) => (dispatch) => {
  return dispatch({
    payload: num,
    type: COUNTER_MINES,
  })
}

