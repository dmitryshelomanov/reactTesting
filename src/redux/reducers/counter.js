import { COUNTER_MINES, COUNTER_PLUS } from '../types'


export const INITIAL_STATE = {
  number: 0
}

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case COUNTER_PLUS: return {
      ...state,
      number: action.payload
    }
    case COUNTER_MINES: return {
      ...state,
      number: action.payload
    } 
    default: return state
  }
}
