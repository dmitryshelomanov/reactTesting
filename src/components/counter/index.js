import React, { Component } from 'react'
import { connect } from 'react-redux'
import FlexWrap from '../ui/flexWrap'
import BlockNumber from '../ui/blockNumber'
import Actionbtn from '../ui/actionBtn'
import { plus, mines } from '../../redux/actions/counter'


export class Counter extends Component {
  render() {
    const { counter, onIncCounter, onDeincCounter } = this.props

    return (
      <FlexWrap>
        <Actionbtn
          text={"+"}  
          onClick={() => onIncCounter(1)}
        />
        <BlockNumber
          number={counter}
        />
        <Actionbtn
          text={"-"}
          onClick={() => onDeincCounter(2)}
        />
      </FlexWrap>
    )
  }
}

export default connect(
  state => ({
    counter: state.counter.number
  }),
  dispatch => ({
    onIncCounter: (num) => { 
      dispatch(plus(num))
    },
    onDeincCounter: (num) => { 
      dispatch(mines(num))
    }
  })
)(Counter)
