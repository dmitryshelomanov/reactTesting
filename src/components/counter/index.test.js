import React from 'react'
import { shallow, render, configure, mount } from 'enzyme'
import sinon from 'sinon'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { plus, mines } from '../../redux/actions/counter'

import { Counter } from './index'
import Actionbtn from '../ui/actionBtn'
import { COUNTER_PLUS, COUNTER_MINES } from '../../redux/types'

configure({ adapter: new Adapter() })

describe('test component Counter', () => {

  it('counter render without crash', () => {
    render(<Counter />)
  })

  it('dispatches plus action', () => {
    const testFn = sinon.stub()
    const wrapper = shallow(
      <Counter
        onIncCounter={() => plus(1)(testFn)}
      />
    )
    wrapper.find(Actionbtn).at(0).simulate('click')
    expect(
      testFn.calledWith({ payload: 1, type: COUNTER_PLUS })
    ).toBeTruthy()
  })

  it('dispatches mines action', () => {
    const testFn = sinon.stub()
    const wrapper = shallow(
      <Counter
        onDeincCounter={() => mines(100)(testFn)}
      />
    )
    wrapper.find(Actionbtn).at(1).simulate('click')
    expect(
      testFn.calledWith({ payload: 100, type: COUNTER_MINES })
    ).toBeTruthy()
  })

  it('shapshot Counter', () => {
    const tree = renderer
      .create(<Counter />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})