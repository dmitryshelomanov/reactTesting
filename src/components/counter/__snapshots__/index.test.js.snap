// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component Counter shapshot Counter 1`] = `
<div
  className="sc-bdVaJa dvorAs"
>
  <div
    className="sc-htpNat ddEKpF"
    onClick={[Function]}
  >
    +
  </div>
  <div
    className="sc-bwzfXH gweSj"
  />
  <div
    className="sc-htpNat ddEKpF"
    onClick={[Function]}
  >
    -
  </div>
</div>
`;
